File: first.py
def search(item,arr):
    for i in range(0,len(arr)):
        if arr[i] == item:
            return i
    return -1

def bsearch(arr, item,start,end):
  if end-start == 1:
    if arr[start] == item:
        return start
    else:
        return -1

  mid = int( (end-start) / 2)

  if arr[start+mid] > item:
    return bsearch(arr, item, start, end-mid)
  else:
    return bsearch(arr, item, start+mid, end)

def insert(item,arr):
    arr.append(item)
    return arr

def delete(item,arr):
    arr.remove(item)
    return arr

arr = [1, 3, 4, 6, 8, 9, 12, 14, 16, 17, 27, 33, 45, 51, 53, 63, 69, 70]
item = 16
print("[+] Index of %s (Binary search): "%(item) + str(bsearch(arr, item,0,len(arr))))

print("[+] Index of %s (Normal search): "%(item) + str(search(item,arr)))

lst = ["a","b","c","d"]

i=insert("e",lst)
print("[+] new arr: ",i)

d=delete("a",lst)
print("[+] nwe arr: ",d)

File: first.cpp
#include <iostream>

using namespace std;
 
int search(int arr[], int n, int x)
{
    int i;
    for (i = 0; i < n; i++)
        if (arr[i] == x)
            return i;
    return -1;
}

int bsearch(int arr[],int l,int r,int x)
{
    if (r >= l) {

        int mid = l + (r - l) / 2;

        if (arr[mid] == x)

            return mid;

        if (arr[mid] > x)

            return bsearch(arr, l, mid - 1, x);
 
        return bsearch(arr, mid + 1, r, x);

    }
    return -1;


}
int main(void)
{

    int arr[] = {1, 3, 4, 6, 8, 9, 12, 14, 16, 17, 27, 33, 45, 51, 53, 63, 69, 70};
    int x = 16;
    int n = sizeof(arr) / sizeof(arr[0]);
    
    int result = search(arr, n, x);
    int bresult = bsearch(arr, 0, n - 1, x);
    
    cout << "[+] index of "<<x<<" (Normal search): "<<result<<endl;
    cout << "[+] index of "<<x<<" (Binary search): "<<bresult;
    return 0;
}
